lib/spack/spack/test/spec_semantics.py

```diff
@@ -985,6 +985,65 @@ def test_forwarding_of_architecture_attributes(self):
         assert 'avx512' not in spec.target
         assert spec.target < 'broadwell'
 
+    @pytest.mark.parametrize('transitive,expected_result', [
+        (True, False),
+        (False, True)
+    ])
+    def test_splice(self, transitive, expected_result):
+        # Tests the new splice function in Spec using a somewhat simple case
+        # with a variant with a conditional dependency.
+        # TODO: Set these two specs to have different targets.
+        # TODO: Test being able to splice in different provider for a virtual.
+        # Example: mvapich for mpich.
+        # TODO: Need to do this with different architectures
+        spec = Spec('splice-t')
+        dep = Spec('splice-h+foo')
+        spec.concretize()
+        dep.concretize()
+        # Sanity checking that these are not the same thing.
+        assert dep.dag_hash() != spec['splice-h'].dag_hash()
+        assert dep.build_hash() != spec['splice-h'].build_hash()
+        # Do the splice.
+        out = spec.splice(dep, transitive)
+        # Returned spec should still be concrete.
+        assert out.concrete
+        # Traverse the spec and assert that the targets are correct.
+        # The following should fail with a "NotFoundError" if the DAGs don't
+        # match by name.
+        for node in spec.traverse():
+            assert node.name == out[node.name].name
+        # Post-splice, dag hash should still be different; no changes should be
+        # made to these specs.
+        assert dep.dag_hash() != spec['splice-h'].dag_hash()
```

Comment:
Make this a separate check that splice doesn't modify in place

Reply:
This new test has been implemented as `test_splice_input_unchanged`.